db/migrate/20200415095034_add_image_upload_id_to_posts_and_topics.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class AddImageUploadIdToPostsAndTopics < ActiveRecord::Migration[6.0]
+  def change
+    add_reference :posts, :image_upload, foreign_key: { to_table: :uploads, on_delete: :nullify }
```

Comment:
A slight concern here I have is that we are adding DB consistency foreign keys in some spots when a whole bunch of columns on the same table do not have them .... like user_id post_id and the list goes on. 

Feels a bit inconsistent as a rule to apply. 

I think we should just stick to adding the column and indexes we want by hand.